assets/sources/source_clients/newegg.py

```diff
@@ -20,7 +20,7 @@ def __init__(self, product_name, product_url, **kwargs):
         )
 
     def out_of_stock(self):
-        return "OUT OF STOCK" in self.soup.find(attrs={"id": "landingpage-stock"}).text
```

Comment:
This was complaining that NoneType(i.e. the data returned from find) was non-iterable. Using find_all seemed to do the trick.

Reply:
So I would need to force an out of stock to test this...but I think what you want here is this

from builtins import any
imgs = self.soup.find_all(attrs={"id": "landingpage-stock"})
return any["OUT OF STOCK" in x for x in img]

But also that makes me wonder? What happens in OUT OF STOCK text is added to another product img that happens to be on that page? This is very curious to me.

Reply:
Implemented this suggestion